sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -305,8 +356,10 @@ public OpenAIAsyncClient buildAsyncClient() {
      *
      * @return an instance of OpenAIClient.
      */
-    @Generated
     public OpenAIClient buildClient() {
+        if (tokenCredential != null && tokenCredential instanceof NonAzureOpenAIKeyCredential) {
+            return new OpenAIClient(buildInnerOpenAIClient());
```

Comment:
What do we do if the endpoint is set but it doesn't point to the public OpenAI service?

Reply:
Since we are using NonAzure Implementation if we find the developer is using NonAzureOpenAIKeyCredential. The endpoint that the developer passed will be ignored because NonAzure Implementation hard-coded endpoint in the class

Reply:
Talked offline. Stay with it.